cluster-sharding/src/main/scala/org/apache/pekko/cluster/sharding/external/ExternalShardAllocationStrategy.scala

```diff
@@ -44,6 +45,21 @@ object ExternalShardAllocationStrategy {
 
   type ShardRegion = ActorRef
 
+  /**
+   * Scala API: Create an [[ExternalShardAllocationStrategy]]
+   */
+  def apply(systemProvider: ClassicActorSystemProvider, typeName: String)(
+      implicit timeout: Timeout = 5.seconds): ExternalShardAllocationStrategy =
+    new ExternalShardAllocationStrategy(systemProvider, typeName)(timeout)
```

Comment:
1. remove the `Scala API` which is very clear this is a `Scala API`.
2. remove the default parameter ` 5.seconds, which is not great for binary compabinity .

Reply:
Let's keep the `Scala API`, otherwise it will be as confused as ActorSystem for Java newcomers.